### PR TITLE
Some small downloader improvements

### PR DESCRIPTION
**Pull request by sschuberth**
Please have a look at the individual commit messages for the details.

### CODE REVIEW

File: scanner/src/funTest/assets/file-counter-expected-output-for-analyzer-result.yml
Comment (fviernau) [inline scanner/src/funTest/assets/file-counter-expected-output-for-analyzer-result.yml:235]: Note that adjusting the exception may break some rule violations.
The upper part of the exception says "No VCS URL provided for".

Considering the above would you agree it makes sense to not do this change?

File: downloader/src/main/kotlin/WorkingTree.kt
Comment (fviernau) [inline downloader/src/main/kotlin/WorkingTree.kt:98]: Why is it needed to have the tags in the exception?

**Review by fviernau — CHANGES_REQUESTED** [review]

File: downloader/src/main/kotlin/WorkingTree.kt
Comment (sschuberth) [reply downloader/src/main/kotlin/WorkingTree.kt:98]: It's not *needed*, but useful to have to debug possible issues in the ORT logic that tries to match version strings to tag names.

File: scanner/src/funTest/assets/file-counter-expected-output-for-analyzer-result.yml
Comment (sschuberth) [reply scanner/src/funTest/assets/file-counter-expected-output-for-analyzer-result.yml:235]: > Note that adjusting the exception may break some rule violations.

Don't you mean error resolutions?

> The upper part of the exception says "No VCS URL provided for".

Indeed, that should probably then also changed to "No VCS provided for" or "No VCS information provided for".

> Considering the above would you agree it makes sense to not do this change?

I admit I forgot to consider the potential impact on error resolutions. (Or really rule violations?) But I'm also not sure how real that case is. Do you have such cases?

File: downloader/src/main/kotlin/WorkingTree.kt
Comment (fviernau) [reply downloader/src/main/kotlin/WorkingTree.kt:98]: From my experience with debugging in this area, I think that debugging effort cannot be significantly improved by adding the tags. E.g. the effort for debugging is huge compared to the effort for listing the remote tags manually. This is why I believe it's not needed. I'll leave that up to you however.

File: scanner/src/funTest/assets/file-counter-expected-output-for-analyzer-result.yml
Comment (fviernau) [reply scanner/src/funTest/assets/file-counter-expected-output-for-analyzer-result.yml:235]: > Don't you mean error resolutions?

I indeed meant error resolutions.

Comment (fviernau) [reply scanner/src/funTest/assets/file-counter-expected-output-for-analyzer-result.yml:235]: > Do you have such cases?

Not in the global config. I can't tell what's in the `ort.yml`s which is why I'm pointing that out.

File: downloader/src/main/kotlin/WorkingTree.kt
Comment (sschuberth) [reply downloader/src/main/kotlin/WorkingTree.kt:98]: Please note that when I say "debug" I don't necessarily mean interactive debugging in the IDE, but the whole process of hunting down a bug, which includes reading DEBUG level logs, stack traces etc.

File: scanner/src/funTest/assets/file-counter-expected-output-for-analyzer-result.yml
Comment (sschuberth) [reply scanner/src/funTest/assets/file-counter-expected-output-for-analyzer-result.yml:235]: OK, I'll drop that change then to be on the safe side.

File: downloader/src/main/kotlin/WorkingTree.kt
Comment (fviernau) [reply downloader/src/main/kotlin/WorkingTree.kt]: Yeah, I used the same for "debug" in this context.

Comment (sschuberth) [reply downloader/src/main/kotlin/WorkingTree.kt]: So, still good for approval / dismissing your review?

**Review by fviernau — APPROVED** [review]

File: downloader/src/main/kotlin/WorkingTree.kt
Comment (fviernau) [reply downloader/src/main/kotlin/WorkingTree.kt]: yes

**Pull request merged**
